src/Chainweb/Pact/SPV.hs

```diff
@@ -311,6 +332,165 @@ extractProof True (Object (ObjectMap o) _ _ _) = case M.lookup "proof" o of
     first (const "Decode of TransactionOutputProof failed") (decodeStrictOrThrow j)
   _ -> Left "Invalid input, expected 'proof' field with base64url unpadded text"
 
+evalHyperlaneCommand :: Object Name -> ExceptT Text IO (Object Name)
+evalHyperlaneCommand (_objectMap . _oObject -> om) = do
+  case (M.lookup "storageLocation" om, M.lookup "signature" om) of
+    (Just (TLitString storageLocation), Just (TLitString sig)) -> recoverAddressValidatorAnnouncement storageLocation sig
+    _ -> case (M.lookup "message" om, M.lookup "metadata" om, M.lookup "validators" om, M.lookup "threshold" om) of
+      (Just (TLitString message), Just (TLitString metadata), Just (TList validators _ _), Just (TLitInteger threshold)) ->
+        let
+          convert (TLitString v) = Just v
+          convert _ = Nothing
+        in verifySignatures message metadata (V.mapMaybe convert validators) (fromInteger threshold)
+
+      (Just (TObject o _), _, _, _) -> encodeHyperMessage o
+      _ -> throwError "Unknown hyperlane command"
+
+verifySignatures :: Text -> Text -> V.Vector Text -> Int -> ExceptT Text IO (Object Name)
+verifySignatures hexMessage hexMetadata validators threshold = do
+  message <- case decodeHex hexMessage of
+          Right s -> pure s
+          Left e -> throwError $ Text.pack $ "Decoding of HyperlaneMessage failed: " ++ e
```

Comment:
ping @edmundnoble: can you double check the throwing strategy here. I don't actually think we've discussed exceptions in `verify-spv` and what to do about them. Ostensibly, they should  kill the tx and we need to make sure they percolate up as pact eval errors.

Reply:
They give an EvalError, I checked Pact. Not a ContError. So `throwError` is the correct thing here to do.